src/status_im/chat/models/message.cljs

```diff
@@ -79,5 +82,276 @@
+                   (dissoc result :db))
+
+      true
+      (as-> fx'
+        (merge fx' (send fx' params')))
+
+      (:to-message command')
+      (assoc :chat-requests/mark-as-answered {:chat-id    chat-id
+                                              :message-id (:to-message command')})
+
+      (= constants/console-chat-id chat-id)
+      (as-> fx'
+        (let [messages (console-events/console-respond-command-messages params' random-id-seq)
+              events   (mapv #(vector :chat-received-message/add %) messages)]
+          (update fx' :dispatch-n into events))))))
+
+(defn invoke-console-command-handler
+  [{:keys [db] :as cofx} {:keys [chat-id command] :as command-params}]
+  (let [fx-fn (get console-events/console-commands->fx (-> command :command :name))
+        result (fx-fn cofx command)]
+    (send-command cofx result chat-id command-params)))
+
+(defn invoke-command-handlers
+  [{{:keys          [bot-db]
+     :accounts/keys [accounts current-account-id]
+     :contacts/keys [contacts] :as db} :db}
+   {{:keys [command params id]} :command
+    :keys [chat-id address]
+    :as orig-params}]
+  (let [{:keys [type name scope-bitmask bot owner-id]} command
+        handler-type (if (= :command type) :commands :responses)
+        to           (get-in contacts [chat-id :address])
+        identity     (or owner-id bot chat-id)
+        bot-db       (get bot-db (or bot chat-id))
+        ;; TODO what's actually semantic difference between `:parameters` and `:context`
+        ;; and do we have some clear API for both ? seems very messy and unorganized now
+        jail-params  {:parameters params
+                      :context    (cond-> {:from            address
+                                           :to              to
+                                           :current-account (get accounts current-account-id)
+                                           :message-id      id}
+                                          (:async-handler command)
+                                          (assoc :orig-params orig-params))}]
+    {:call-jail {:jail-id                 identity
+                 :path                    [handler-type [name scope-bitmask] :handler]
+                 :params                  jail-params
+                 :callback-events-creator (fn [jail-response]
+                                            [(when-not (:async-handler command)
+                                               [:command-handler! chat-id orig-params jail-response])])}}))
+
+(defn process-command
+  [{:keys [db] :as cofx} {:keys [command message chat-id] :as params}]
+  (let [{:keys [command] :as content} command]
+    (-> {:db (chat-model/set-chat-ui-props db {:sending-in-progress? false})}
+
+        (as-> fx'
+          (cond
+            (and (= constants/console-chat-id chat-id)
+                 (console-events/commands-names (:name command)))
+            (invoke-console-command-handler (merge cofx fx') params)
```

Comment:
Why merging cofx with fx ? Seems quite confusing and dangerous (easy overwrite/pass something you don't want), as the `invoke-command-handlers` actually needs only `:db` coeffect and nothing else, it would be much better to change the function to take db only and call it with `(invoke-console-command-handler (:db fx'))`, the same is true for `invoke-command-handlers`

Reply:
Have you looked at how `invoke-console-command-handler` function looks like and what's happening inside of it?

Reply:
But in short words, no, it needs more than just `db` and it's not dangerous because there is no way to overwrite something we don't want in this case.

Reply:
Still, I think it's much clearer to treat `coeffects` and `effects` separately and name them such, rather then merging them together (make the function take one extra arg if necessary).

Reply:
Ok, fixed it.